control/veda_val.py

```diff
@@ -356,9 +356,12 @@ def send_200(self, val_api_return):
         VAL will not allow duped studio urls to be sent,
         so we must scrub the data
         """
-        for c in val_api_return['courses']:
-            if c in self.val_data['courses']:
-                self.val_data['courses'].remove(c)
+        for course in val_api_return['courses']:
+            for key, entry in course.items():
+                for crs in self.val_data['courses']:
```

Comment:
What, conceptually, are the things in `self.val_data['courses']`? Are they existing courses?  If so, maybe a better  variable name for `crs` is `existing_course`, or something like that.